src/internal/v1api/modules.go

```diff
@@ -72,24 +73,8 @@ func (g Generator) readModuleMetadata(path string, logger *slog.Logger) (*module
 // writeModuleVersionListing writes the file containing the module version listing.
 // This data  is to be consumed when an end user requests /v1/modules/{namespace}/{name}/{targetSystem}/versions
 func (g Generator) writeModuleVersionListing(namespace string, name string, targetSystem string, versions []ModuleVersionResponseItem) error {
-	destinationDir := filepath.Join(g.DestinationDir, "v1", "modules", namespace, name, targetSystem)
-	if err := os.MkdirAll(destinationDir, 0755); err != nil {
-		return fmt.Errorf("failed to create directory: %w", err)
-	}
-
-	filePath := filepath.Join(destinationDir, "versions")
-
-	marshalled, err := json.Marshal(ModuleVersionListingResponse{Modules: []ModuleVersionListingResponseItem{{Versions: versions}}})
-	if err != nil {
-		return fmt.Errorf("failed to marshal json: %w", err)
-	}
-
-	err = os.WriteFile(filePath, marshalled, 0644)
-	if err != nil {
-		return fmt.Errorf("failed to write file: %w", err)
-	}
-
-	return nil
+	path := filepath.Join(g.DestinationDir, "v1", "modules", namespace, name, targetSystem, "versions")
+	return files.WriteToJsonFile(path, ModuleVersionListingResponse{Modules: []ModuleVersionListingResponseItem{{Versions: versions}}})
```

Comment:
If we are keeping this method call the way it is, I would rather explicitly handle the error here too instead of returning directly. 

this allows us to wrap the error nicely and provide information about what failed and where.